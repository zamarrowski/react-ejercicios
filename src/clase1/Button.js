import PropTypes from 'prop-types'

const Button = props => (
  <button onClick={props.onClick} className="btn">
    {props.label}
  </button>
)

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
}

Button.defaultProps = {
  label: 'Default text',
}

export default Button
