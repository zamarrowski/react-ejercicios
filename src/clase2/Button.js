import styled, { css } from 'styled-components'

const Button = styled.button`
  padding: 1rem;
  border: 1px solid blue;
  background-color: ${props => props.theme.main};

  ${props =>
    props.success &&
    css`
      border-color: green;
    `}

  ${props =>
    props.danger &&
    css`
      border-color: red;
    `}
`

export const TomatoButton = styled(Button)`
  background-color: tomato;
`

export default Button
