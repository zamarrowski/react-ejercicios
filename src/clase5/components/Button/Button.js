import styled, { css } from 'styled-components'

const Button = styled.button`
  padding: 0.5rem 1rem;
  background-color: greenyellow;

  ${props =>
    props.danger &&
    css`
      background-color: red;
      color: white;
    `}
`

export default Button
