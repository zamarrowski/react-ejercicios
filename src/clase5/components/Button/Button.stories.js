import Button from './Button'

export default {
  title: 'Button',
  component: Button,
}

export const Default = () => <Button>hola</Button>
export const Danger = () => <Button danger>hola</Button>
