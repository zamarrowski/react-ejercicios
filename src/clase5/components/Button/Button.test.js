import { render, screen } from '@testing-library/react'
import Button from './Button'

test('Should render without errors', () => {
  render(<Button />)
})

test('Should match snapshot', () => {
  render(<Button>test</Button>)
  expect(screen.getByRole('button')).toMatchSnapshot()
})
