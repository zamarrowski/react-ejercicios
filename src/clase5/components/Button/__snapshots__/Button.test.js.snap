// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  background-color: greenyellow;
}

<button
  class="c0"
>
  test
</button>
`;
